### DISCUSSION TITLE
Problem with crosspromotion.swc can't compile successful with swc with Harman Air

### DISCUSSION THREAD

**Discussion by DeafMan1983** [prominic/Moonshine-IDE#973] [General]
Hello everyone, I would like to tell why do I can't build swc with Air package like I want wrapper with air native extension

See in [Adobe Air Github](https://github.com/airsdk/Adobe-Runtime-Support/discussions/1643). But I don't know which discussion.

I am very disappointed because I don't have AIR SDK 26.0 because it has crosspromotion.swc since AiR SDK 27.0 to 33.1 don't have crosspromotion.swc

Thanks!

**Comment by piotrzarzycki21** [comment]
@DeafMan1983 I'm not sure if this issue is related to Moonshine IDE itself. I saw in original issue raised on AIR Github that you are trying to use instellr.xml to download Air 33.1. Please try our Moonshine [SDK Installer Nightly build](https://moonshine-ide.com/download-sdk-installer/) where you have option to download Flex + Air 33.1 (scroll to the bottom of page)

**Comment by DeafMan1983** [comment]
Hey @piotrzarzycki21 Oh yes I already downloaded and tested but Flex SDK with Air SDK 33.1 and it throws error exception. It is same why do I not have crosspromotion.swc?

I have proof 
Look [youtube](https://www.youtube.com/watch?v=t2yi_qGvhOY)!
in video I explain I already installed MoonShineSDK Installer and remove Flex- and Air-SDK and redownload again but It doesn't have crosspromotion.swc

Is MoonShine throw crosspromotion.swc that is werid cause it looks like it has bug or issue?

// EDIT:
Wow! I found because MoonShine has bug because it doesn't find crosspromotion.swc cause it looks irrelavantly.

I tried with Visual Studio Code and I create simple Library and it compiled successfully.
Look [Youtube](https://www.youtube.com/watch?v=-sndg2Cfvts). Why does MoonShine not compile for librarry if I use with configuration Air SDK then it throws error exception with crosspromotion.swc 
I believe MoonShine has bug. PS: Please remove crosspromotion.swc from external library-paths!

**Comment by joshtynjala** [comment]
FYI — Commit 1d34e3fb7fd193176c01464baf2c50e0946a36fa should fix this issue in the next future update to Moonshine.

**Comment by piotrzarzycki21** [comment]
Thanks! I have created issue just to have this documented -> https://github.com/Moonshine-IDE/Moonshine-IDE/issues/1068
